BVTCK-57 Followup: Avoid the use of @BeforeMethod

marko-bekhta opened the pull request:
It's a follow up on 

- https://hibernate.atlassian.net/browse/BVTCK-57

As @gunnarmorling mentioned in the comments of https://github.com/beanvalidation/beanvalidation-tck/pull/86#issuecomment-287355830 I've added checkstyle plugin with configs for imports only. Turned out that there were a few violations so I've cleaned those up. 

Also with that last commit I've removed the last `Before/After`'s they were used to change `Locale`.

As enabling other checlstyle rules will require some code modifications (for example `NewlineAtEndOfFile` is failing in each file :) ), I think it'll be better if done separately from this one.
Let me know if that would be useful and if so - I can work on it :)

gsmet commented on tests/src/main/java/org/hibernate/beanvalidation/tck/tests/messageinterpolation/ExpressionLanguageMessageInterpolationTest.java:
```diff
@@ -241,4 +253,33 @@ public boolean isValid(Continent continent, ConstraintValidatorContext constrain
+		}
+
+		public static void test(TestLogic testLogic) {
+			new TestWrapper( testLogic, Locale.ENGLISH ).doTest();
```
This is weird. You push the ENGLISH locale as the testLocale every time, while it might not be indeed. But if it's not, you use `Locale.setDefault` in the `TestLogic`.

I would rather add a second `test` method taking an additional locale parameter and use this locale for the test. It looks like it's what you wanted to do but decided otherwise in the way :).

I would even rename the method `testWithEnglishLocale` and `testWithSpecificLocale` to be more explicit.

Or am I missing something?

gsmet replied:
@marko-bekhta we discussed it with Gunnar and we think it's better to simply disable checkstyle for these imports in this test as the usage is legitimate here (see https://github.com/beanvalidation/beanvalidation-tck/pull/91#issuecomment-287716857 ).

The test will be more readable this way.

marko-bekhta replied:
@gsmet great, thanks for the update! I'll rework the last commit to ignore this import rules for this test

gsmet commented:
@gunnarmorling the issue I see with using checkstyle to avoid the usage of these 2 annotations is that they indeed might be used correctly. Typically, in the locale case, I don't think it's an issue to use them. I don't foresee a lot more of these correct usages though so we might consider this limitation acceptable.

gunnarmorling commented:
> they indeed might be used correctly. Typically, in the locale case, I don't think it's an issue to use them

One can turn off CS for specific code sections via comments (`//CHECKSTYLE.OFF ... //CHECKSTYLE.ON`). It's a bit verbose, but I think it's a good trade-off as there should be just a few legitimate uses.

That said, is setting the locale on the client-side actually a legitimate use? Shouldn't it be done within the VM running the tests?

gsmet commented:
> That said, is setting the locale on the client-side actually a legitimate use? Shouldn't it be done within the VM running the tests?

It sure can. It's probably more correct but also more verbose. I think Marko found a good trade-off (once my remarks are applied) so I would lean towards accepting it. Just wanted to point this out.

gunnarmorling commented:
This one still needs more time to be updated to add back `@BeforeMethod` and add //CHECKSTYLE-OFF for the few locale uses, right?

marko-bekhta commented:
@gunnarmorling that's right. I'll be doing this a bit later today.

marko-bekhta commented:
I've reworked last commit and added `SuppressionCommentFilter` and commented out those two imports with `//CHECKSTYLE-OFF //CHECKSTYLE-ON`. So this PR seems to be complete now.

As for the other checkstyle rules - would it be of use enabling them and making the code comply those rules?

gunnarmorling commented:
> So this PR seems to be complete now.

Awesome, rebased and applied. Thanks a lot, Marko!

> As for the other checkstyle rules - would it be of use enabling them

Not needed from my side. I'm no CheckStyle zealot really. @gsmet?
